controllers/metal3.io/preprovisioningimage_controller.go

```diff
@@ -351,9 +353,22 @@ func (r *PreprovisioningImageReconciler) CanStart() bool {
 	return false
 }
 
-func (r *PreprovisioningImageReconciler) SetupWithManager(mgr ctrl.Manager) error {
+func (r *PreprovisioningImageReconciler) SetupWithManager(mgr ctrl.Manager, options controller.Options) error {
+	maxConcurrentReconciles := 0
+	if options.MaxConcurrentReconciles > 0 {
+		ctrl.Log.Info(fmt.Sprintf("Controller Concurrency will be set to %d according to command line flag", options.MaxConcurrentReconciles))
+	} else {
+		mCR, err := getMaxConcurrentReconciles()
+		if err != nil {
+			return err
+		}
+		maxConcurrentReconciles = mCR
+	}
+	options.MaxConcurrentReconciles = maxConcurrentReconciles
```

Comment:
I think you need to place this line instead of line 365. Otherwise you're setting options.MaxConcurrentReconciles to 0 if options.MaxConcurrentReconciles > 0.